lib/time.ml

```diff
@@ -51,13 +51,13 @@ type sleep = {
 }
 
 module SleepQueue =
-  Lwt_pqueue.Make (struct
+  Binary_heap.Make (struct
                      type t = sleep
-                     let compare { time = t1; _ } { time = t2; _ } = compare t1 t2
+                     let compare { time = t1; _ } { time = t2; _ } = compare t2 t1
```

Comment:
What's the reason for changing `compare t1 t2` to `compare t2 t1`? I don't actually know these APIs, so I'm not commenting on correctness, it just jumped out on me as an "odd" change.

Reply:
In the former `Lwt_pqueue`, you were allowed to get the minimum of the elements in the priority queue.
`Binary_heap` operates on maxmiums instead, so in order to keep the same semantics, I had to change the `compare` function so that minimums become maximums.

Reply:
To make this look a bit less weird (and make intent more explicit), how about something like that:

```ocaml
(** X library only operates on maximums, but we need the minimum,
    so invert the order *)
let invert_comparison x = -x

let compare { time = t1; ... } { time = t2; ... } =
    invert_comparison (compare t1 t2)
```

Reply:
rather than "less weird", let's say "looks less like a typo"

Reply:
Sounds good

Reply:
Yes please!

Reply:
:heavy_check_mark: Done